rust-workspace/crates/ntc-asset-manager/src/algorand.rs

```diff
@@ -0,0 +1,126 @@
+use crate::Secret;
+
+/// Aliased re-export.
+pub use algonaut::error::AlgonautError;
+/// Aliased re-export.
+pub use algonaut::{
+    core::Address as AlgorandAddress,
+    model::algod::v2::TransactionResponse as AlgorandTransactionResponse,
+    transaction::{
+        error::TransactionError as AlgorandTransactionError, CreateAsset as AlgorandCreateAsset,
+        SignedTransaction as AlgorandSignedTransaction, Transaction as AlgorandTransaction,
+    },
+};
+/// Aliased re-export.
+pub use algonaut_client::error::ClientError as AlgonautClientError;
+
+// Import *real* struct when building
+#[cfg(not(test))]
+use algonaut::transaction::account::Account as AlgorandAccount;
+
+// Import *mock* struct when testing
+#[cfg(test)]
+use tests::MockAlgorandAccount as AlgorandAccount;
+
+use thiserror::Error;
+
+#[derive(Error, Debug)]
+pub enum AlgorandSigningError {
+    #[error("conversion error: {0:?}")]
+    ConversionError(#[from] std::convert::Infallible),
```

Comment:
Discussed in Discord, this error can be removed. Maybe the enum is not needed if `TransactionError` is the only enum type.